src/lib/date.spec.ts

```diff
@@ -38,7 +82,30 @@ describe("date tests", () => {
     [new Date(2014, 3, 15, 23, 59), new Date(2014, 3, 16, 0, 0), 1],
     [null as unknown as Date, new Date(), Number.NaN],
     [new Date(), null as unknown as Date, Number.NaN],
+    [undefined as unknown as Date, new Date(), Number.NaN],
+    [new Date(), undefined as unknown as Date, Number.NaN],
+    [42 as unknown as Date, new Date(), Number.NaN],
+    [new Date(), 42 as unknown as Date, Number.NaN],
+    ["2014-03-15" as unknown as Date, new Date(), Number.NaN],
+    [new Date(), "2014-03-15" as unknown as Date, Number.NaN],
   ])("dateDifferenceInDays", (dateFrom, dateTo, expected) => {
     expect(dateDifferenceInDays(dateFrom, dateTo)).toBe(expected);
   });
+
+  test.each([
+    [new Date(2024, 0, 1), new Date(2024, 0, 1)],
+    [new Date(2024, 0, 2), new Date(2024, 0, 2)],
+    [new Date(2024, 0, 3), new Date(2024, 0, 3)],
+    [new Date(2024, 0, 4), new Date(2024, 0, 4)],
+    [new Date(2024, 0, 5), new Date(2024, 0, 5)],
+    [new Date(2024, 0, 6), new Date(2024, 0, 8)], // Weekend
+    [new Date(2024, 0, 7), new Date(2024, 0, 8)], // Weekend
+    [new Date(2024, 0, 8), new Date(2024, 0, 8)],
+    [null as unknown as Date, null],

```

Comment:
Are we really sure that we don't want to have a precise return type in case of an invalid date is provided? 

I would go for always return undefined/null or a throw error personally

Reply:
@neotrow what is your opinion about this?
does it make sense to return something if the argument is not a valid date?
Maybe is better to go with the safer approach and throw an exception?

PS: in date-fns if the return type is a Date an the arguments are not valid it returns `new Date(NaN)`, but I'm not sure about this solution either

Reply:
as by definition for js, a date cannot be null, so from my point of view we shouldn't return a null. most other languages would throw an error when you pass a corrupt object, so i would probably also throw one here 🤔

Reply:
very sorry for the late reply @drebrez 

I agree with @neotob that null is not a valid date in JS so that's the wrong appraoch IMO. 

I don't have a strong opinion about throwing an exception or returning `new Date(NaN)`.